<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Mainwindow</class>
 <widget class="QDialog" name="Mainwindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1783</width>
    <height>919</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>3D_Viewer_v2</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(119, 118, 123);
border-top-color: rgb(36, 31, 49);</string>
  </property>
  <widget class="s21::glViewer" name="widget" native="true">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>891</width>
     <height>851</height>
    </rect>
   </property>
  </widget>
  <widget class="QComboBox" name="projection">
   <property name="geometry">
    <rect>
     <x>960</x>
     <y>170</y>
     <width>171</width>
     <height>51</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QComboBox {color: rgb(211, 213, 201); background-color: rgb(36, 36, 44);}</string>
   </property>
   <item>
    <property name="text">
     <string>parallel</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>central</string>
    </property>
   </item>
  </widget>
  <widget class="QPushButton" name="file_add">
   <property name="geometry">
    <rect>
     <x>1420</x>
     <y>40</y>
     <width>221</width>
     <height>81</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton {color: rgb(211, 213, 201); background-color: rgb(36, 36, 44);}</string>
   </property>
   <property name="text">
    <string>Add file</string>
   </property>
  </widget>
  <widget class="QTextBrowser" name="info_box">
   <property name="geometry">
    <rect>
     <x>960</x>
     <y>30</y>
     <width>431</width>
     <height>111</height>
    </rect>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>1160</x>
     <y>740</y>
     <width>221</width>
     <height>81</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_scale">
    <item>
     <widget class="QSlider" name="horizontalSlider_scale">
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>20</number>
      </property>
      <property name="singleStep">
       <number>1</number>
      </property>
      <property name="value">
       <number>1</number>
      </property>
      <property name="sliderPosition">
       <number>1</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="tickInterval">
       <number>1</number>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="label_scale">
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string> Scale</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="label_value_scale">
      <property name="minimumSize">
       <size>
        <width>30</width>
        <height>0</height>
       </size>
      </property>
      <property name="baseSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>1</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>960</x>
     <y>580</y>
     <width>601</width>
     <height>164</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_transformer">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_rotate">
      <property name="spacing">
       <number>22</number>
      </property>
      <property name="topMargin">
       <number>5</number>
      </property>
      <property name="bottomMargin">
       <number>5</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_rotate_x">
        <item>
         <widget class="QLabel" name="label_rotate_x">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string> X rotate </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="horizontalSlider_rotate_x">
          <property name="minimum">
           <number>-180</number>
          </property>
          <property name="maximum">
           <number>180</number>
          </property>
          <property name="singleStep">
           <number>1000</number>
          </property>
          <property name="pageStep">
           <number>1000</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="tickInterval">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_value_rotate_x">
          <property name="minimumSize">
           <size>
            <width>30</width>
            <height>0</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>0</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_rotate_y">
        <item>
         <widget class="QLabel" name="label_rotate_y">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string> Y rotate </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="horizontalSlider_rotate_y">
          <property name="minimum">
           <number>-180</number>
          </property>
          <property name="maximum">
           <number>180</number>
          </property>
          <property name="singleStep">
           <number>100</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_value_rotate_y">
          <property name="minimumSize">
           <size>
            <width>30</width>
            <height>0</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>0</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_rotate_z">
        <item>
         <widget class="QLabel" name="label_rotate_z">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string> Z rotate </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="horizontalSlider_rotate_z">
          <property name="minimum">
           <number>-180</number>
          </property>
          <property name="maximum">
           <number>180</number>
          </property>
          <property name="singleStep">
           <number>100</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_value_rotate_z">
          <property name="minimumSize">
           <size>
            <width>30</width>
            <height>0</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>0</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_translate">
      <property name="spacing">
       <number>22</number>
      </property>
      <property name="topMargin">
       <number>5</number>
      </property>
      <property name="bottomMargin">
       <number>5</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_translate_x">
        <item>
         <widget class="QLabel" name="label_translate_x">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string> X translate</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="horizontalSlider_translate_x">
          <property name="minimum">
           <number>-50</number>
          </property>
          <property name="maximum">
           <number>50</number>
          </property>
          <property name="singleStep">
           <number>1</number>
          </property>
          <property name="pageStep">
           <number>10</number>
          </property>
          <property name="value">
           <number>0</number>
          </property>
          <property name="tracking">
           <bool>true</bool>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="tickPosition">
           <enum>QSlider::NoTicks</enum>
          </property>
          <property name="tickInterval">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_value_translate_x">
          <property name="minimumSize">
           <size>
            <width>30</width>
            <height>0</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>0</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_translate_y">
        <item>
         <widget class="QLabel" name="label_translate_y">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string> Y translate</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="horizontalSlider_translate_y">
          <property name="minimum">
           <number>-50</number>
          </property>
          <property name="maximum">
           <number>50</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_value_translate_y">
          <property name="minimumSize">
           <size>
            <width>30</width>
            <height>0</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>0</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_translate_z">
        <item>
         <widget class="QLabel" name="label_translate_z">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string> Z translate</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="horizontalSlider_translate_z">
          <property name="minimum">
           <number>-50</number>
          </property>
          <property name="maximum">
           <number>50</number>
          </property>
          <property name="value">
           <number>-5</number>
          </property>
          <property name="sliderPosition">
           <number>-5</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="tickInterval">
           <number>100</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_value_translate_z">
          <property name="minimumSize">
           <size>
            <width>30</width>
            <height>0</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>0</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QPushButton" name="gif">
   <property name="geometry">
    <rect>
     <x>1640</x>
     <y>170</y>
     <width>91</width>
     <height>291</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton {color: rgb(211, 213, 201); background-color: rgb(36, 36, 44);}</string>
   </property>
   <property name="text">
    <string>gif</string>
   </property>
  </widget>
  <widget class="QPushButton" name="screen">
   <property name="geometry">
    <rect>
     <x>1640</x>
     <y>520</y>
     <width>91</width>
     <height>291</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton {color: rgb(211, 213, 201); background-color: rgb(36, 36, 44);}</string>
   </property>
   <property name="text">
    <string>screen</string>
   </property>
  </widget>
  <widget class="QWidget" name="gridLayoutWidget">
   <property name="geometry">
    <rect>
     <x>960</x>
     <y>250</y>
     <width>601</width>
     <height>181</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="1" column="0">
     <widget class="QLabel" name="Size_t">
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>Size:</string>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QComboBox" name="edge_type">
      <property name="styleSheet">
       <string notr="true">QComboBox {color: rgb(211, 213, 201); background-color: rgb(36, 36, 44);}</string>
      </property>
      <item>
       <property name="text">
        <string>solid</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>dashed</string>
       </property>
      </item>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="thickn_t">
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>Thickness:</string>
      </property>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QComboBox" name="vertex_type">
      <property name="styleSheet">
       <string notr="true">QComboBox {color: rgb(211, 213, 201); background-color: rgb(36, 36, 44);}

</string>
      </property>
      <item>
       <property name="text">
        <string>circle</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>square</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>none</string>
       </property>
      </item>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QSlider" name="edge_thickness">
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QSlider" name="vetex_size">
      <property name="minimumSize">
       <size>
        <width>211</width>
        <height>32</height>
       </size>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="gridLayoutWidget_2">
   <property name="geometry">
    <rect>
     <x>960</x>
     <y>470</y>
     <width>601</width>
     <height>80</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="3">
     <widget class="QPushButton" name="setEdg">
      <property name="styleSheet">
       <string notr="true">QPushButton {color: rgb(211, 213, 201); background-color: rgb(36, 36, 44);}</string>
      </property>
      <property name="text">
       <string>set edges</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QPushButton" name="setBackgr">
      <property name="styleSheet">
       <string notr="true">QPushButton {color: rgb(211, 213, 201); background-color: rgb(36, 36, 44);}</string>
      </property>
      <property name="text">
       <string>set background</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QPushButton" name="setVert">
      <property name="styleSheet">
       <string notr="true">QPushButton {color: rgb(211, 213, 201); background-color: rgb(36, 36, 44);}</string>
      </property>
      <property name="text">
       <string>set verticies</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QPushButton" name="pushButton">
   <property name="geometry">
    <rect>
     <x>1640</x>
     <y>830</y>
     <width>91</width>
     <height>51</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton {color: rgb(211, 213, 201); background-color: rgb(36, 36, 44);}</string>
   </property>
   <property name="text">
    <string>default</string>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>s21::glViewer</class>
   <extends>QWidget</extends>
   <header>glViewer.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
